### PR TITLE
Add newsboat cleaner

### PR DESCRIPTION
**Pull request by dngray**
This one is simple enough it should be good to go. On a side note my other ones I submitted have been working perfectly for months and could go into release.

- [Lftp](https://github.com/az0/cleanerml/blob/master/pending/lftp.xml)
- [NeoVim](https://github.com/az0/cleanerml/blob/master/pending/nvim.xml)
- [Wget](https://github.com/az0/cleanerml/blob/master/pending/wget.xml)
- [Zathura](https://github.com/az0/cleanerml/blob/master/pending/zathura.xml)

### CODE REVIEW

**Comment by Minoru** [comment]
Hi! Newsboat maintainer here.

I skimmed BleachBit's site and it seems that the purpose of the tool is to remove files that aren't strictly necessary, like caches, histories and such. If that's correct, I'd advise against removing ~/.newsboat/cache.db. Even though it's named "cache", it's actually user's data—fetched articles, their flags, read/unread status, etc.

BleachBit's site mentions that the tool can vacuum Firefox. Newsboat's cache.db is an SQLite database too, so you can apply the same machinery to it as well, to save a bit of space. You can do it directly with SQLite, or via Newsboat by calling `newsboat --vacuum`.

Finally, please note that Newsboat can store its files in either the dotdir (~/.newsboat) or in XDG dirs ($XDG_CONFIG_HOME/newsboat, $XDG_DATA_HOME/newsboat). This PR only cleans the former. Be careful not to *create* any of those, as Newsboat checks for their existence to figure out which ones to use.

**Comment by dngray** [comment]
> Hi! Newsboat maintainer here.

Fancy seeing you here.

> I skimmed BleachBit's site and it seems that the purpose of the tool is to remove files that aren't strictly necessary, like caches, histories and such. If that's correct, I'd advise against removing ~/.newsboat/cache.db. Even though it's named "cache", it's actually user's data—fetched articles, their flags, read/unread status, etc.
> 
> BleachBit's site mentions that the tool can vacuum Firefox. Newsboat's cache.db is an SQLite database too, so you can apply the same machinery to it as well, to save a bit of space. You can do it directly with SQLite, or via Newsboat by calling `newsboat --vacuum`.

I might then see if it's possible to vacuum that then.

> Finally, please note that Newsboat can store its files in either the dotdir (~/.newsboat) or in XDG dirs ($XDG_CONFIG_HOME/newsboat, $XDG_DATA_HOME/newsboat). This PR only cleans the former. Be careful not to _create_ any of those, as Newsboat checks for their existence to figure out which ones to use.

Oh, I thought it only stored in ~/.newsboat to be honest I prefer $XDG_CONFIG_HOME, therefore I will make sure to check all three places.

File: pending/newsboat.xml [hunk 1]
```diff
@@ -25,17 +25,23 @@
...
+    <action command="delete" search="file" path="$XDG_CONFIG_HOME/newsboat/history.search"/>
+    <action command="delete" search="file" path="$XDG_DATA_HOME/newsboat/history.search"/>
   </option>
   <option id="cmdline_history">
     <label>Delete the command line history</label>
     <description>Clears the previous command history</description>
-    <action command="delete" search="file" path="~/.newsboat/history.cmdline"/>
+    <action command="delete" search="file" path="$HOME/.newsboat/history.cmdline"/>
+    <action command="delete" search="file" path="$XDG_CONFIG_HOME/newsboat/history.search"/>
+    <action command="delete" search="file" path="$XDG_DATA_HOME/newsboat/history.search"/>
```
Comment (az0) [inline pending/newsboat.xml]: Lines 45 and 45 look like 37 and 38. Is the repetition intentional?

Comment (dngray) [reply pending/newsboat.xml]: My bad, that's what happens when you're missing the :zzz:.

**Pull request merged**